src/test/java/roomescape/dao/ReservationDaoTest.java

```diff
@@ -0,0 +1,179 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.JdbcTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.jdbc.Sql;
+import roomescape.dao.condition.ReservationInsertCondition;
+import roomescape.dao.condition.ThemeInsertCondition;
+import roomescape.dao.condition.TimeInsertCondition;
+import roomescape.domain.Reservation;
+import roomescape.domain.ReservationTheme;
+import roomescape.domain.ReservationTime;
+
+@JdbcTest
+@Import({ReservationDao.class, ReservationTimeDao.class, ReservationThemeDao.class})
+@Sql(scripts = {"/test_schema.sql"})
+public class ReservationDaoTest {
+
+    @Autowired
+    private ReservationDao reservationDao;
+
+    @Autowired
+    private ReservationTimeDao timeDao;
+
+    @Autowired
+    private ReservationThemeDao themeDao;
+
+    @BeforeEach
+    void setUp() {
+        timeDao.insert(new TimeInsertCondition(LocalTime.parse("10:00")));
+        themeDao.insert(new ThemeInsertCondition("theme1", "desc1", "thumb1"));
+    }
+
+    @DisplayName("예약을 추가한다.")
+    @Test
+    void insert() {
+        // given
+        Reservation inserted = reservationDao.insert(new ReservationInsertCondition(
+                "name",
+                LocalDate.parse("2025-01-01"),
+                new ReservationTime(1L, LocalTime.parse("10:00")),
+                new ReservationTheme(1L, "theme_name", "desc", "thumb"))
+        );
+
+        // when
+        Long insertedReservationId = inserted.getId();
+
+        // then
+        assertThat(insertedReservationId).isEqualTo(1L);
+    }
+
+    @DisplayName("모든 예약을 조회한다.")
+    @Test
+    void findAll() {
+        // given
+        reservationDao.insert(new ReservationInsertCondition(
+                "name",
+                LocalDate.parse("2025-01-01"),
+                new ReservationTime(1L, LocalTime.parse("10:00")),
+                new ReservationTheme(1L, "theme_name", "desc", "thumb"))
+        );
+
+        // when
+        List<Reservation> reservations = reservationDao.findAll();
+
+        // then
+        assertThat(reservations.size()).isEqualTo(1);
+    }
+
+    @DisplayName("예약 ID를 이용하여 예약을 조회한다.")
+    @Test
+    void findById() {
+        // given
+        reservationDao.insert(new ReservationInsertCondition(
+                "name",
+                LocalDate.parse("2025-01-01"),
+                new ReservationTime(1L, LocalTime.parse("10:00")),
+                new ReservationTheme(1L, "theme_name", "desc", "thumb"))
+        );
+
+        // when
+        Reservation reservation = reservationDao.findById(1L).orElseThrow();
+
+        // then
+        assertThat(reservation.getName()).isEqualTo("name");
+    }
+
+    @DisplayName("예약을 조회할 때 ID가 존재하지 않으면 빈 예약을 반환한다.")
+    @Test
+    void findByWrongId() {
+        Optional<Reservation> reservation = reservationDao.findById(1L);
+
+        assertThat(reservation).isEqualTo(Optional.empty());
```

Comment:
아래와 같이 간소화 할 수 있으니 참고하시면 좋을것 같습니다.

```suggestion
        assertThat(reservation).isEmpty();
```